src/js/utils/log.js

```diff
@@ -76,32 +83,124 @@ export const logByType = (type, args, stringify = !!IE_VERSION && IE_VERSION < 1
 };
 
 /**
- * Log plain debug messages
+ * Logs plain debug messages. Similar to `console.log`.
  *
- * @param {Mixed[]} args
- *        One or more messages or objects that should be logged.
+ * @class
+ * @param    {Mixed[]} args
+ *           One or more messages or objects that should be logged.
  */
 log = function(...args) {
   logByType('log', args);
 };
 
 /**
- * Keep a history of log messages
+ * Retains a history of all arguments passed to logging functions.
+ *
+ * These records are retained regardless of logging level.
+ *
+ * Whether logging is enabled or not, users would be wise to think carefully
+ * about what they pass to video.js' logging functions. Because values passed
+ * are stored here, they could lead to memory leaks.
  *
  * @type {Array}
  */
 log.history = [];
 
 /**
- * Log error messages
+ * Enumeration of available logging levels, where the keys are the level names
+ * and the values are `|`-separated strings containing logging methods allowed
+ * in that logging level. These strings are used to create a regular expression
+ * matching the function name being called.
+ *
+ * Levels provided by video.js are:
+ *
+ * - `off`: Matches no calls. Any value that can be cast to `false` will have
+ *   this effect. The most restrictive.
+ * - `all` (default): Matches only Video.js-provided functions (`log`,
+ *   `log.warn`, and `log.error`).
+ * - `log`: Matches `log` calls only.
+ * - `warn`: Matches `log.warn` and `log.error` calls.
+ * - `error`: Matches only `log.error` calls.
+ *
+ * @type {Object}
+ */
+log.levels = {};
+
+if (Object.defineProperties) {
+  Object.defineProperties(log.levels, {
+    all: {
+      enumerable: true,
+      value: 'log|warn|error'
```

Comment:
ahh ok, just wanted to make sure the more traditional system was considered, but I do see the benefits here.